wandb/sync.py

```diff
@@ -397,6 +399,7 @@ def stop(self):
                 'Script ended because of Exception, press ctrl-c to abort syncing.')
         else:
             wandb.termlog('Script ended.')
+        self._system_stats.flush()
```

Comment:
flush() might be called twice in parallel (once here and once in thread body), which seems bad. Better would be to have a shutdown command that signals the thread while loop to exit (by setting a boolean on the SystemStats instance, or using a queue). See how "finished" works in FileStreamApi. You should also then wait til it's done shutting down, to prevent the main thread and program from exiting while SystemStats is still flushing.

Reply:
A quick fix for OpenAI would be to comment out this line for now.